src/coverlet.core/CoverageResult.cs

```diff
@@ -196,33 +196,26 @@ public ThresholdTypeFlags GetThresholdTypesBelowThreshold(CoverageSummary summar
                     break;
                 case ThresholdStatistic.Average:
                     {
-                        double line = 0;
-                        double branch = 0;
-                        double method = 0;
+                        var line = summary.CalculateLineCoverage(Modules).AverageModulePercent;
+                        var branch = summary.CalculateBranchCoverage(Modules).AverageModulePercent;
+                        var method = summary.CalculateMethodCoverage(Modules).AverageModulePercent;
                         int numModules = Modules.Count;
 
-                        foreach (var module in Modules)
-                        {
-                            line += summary.CalculateLineCoverage(module.Value).Percent;
-                            branch += summary.CalculateBranchCoverage(module.Value).Percent;
-                            method += summary.CalculateMethodCoverage(module.Value).Percent;
-                        }
-
                         if ((thresholdTypes & ThresholdTypeFlags.Line) != ThresholdTypeFlags.None)
                         {
-                            if ((line / numModules) < threshold)
+                            if (line < threshold)
```

Comment:
Since we aren't not rounding down average value of `line`, there is chance that we will output coverage failed for average threshold but display value indicate it shouldn't have failed.

Ex: threshold value is "85.25" and lets say `line` value is "85.2463", then if condition will evaluate to true and we output coverage failed for average threshold. But while displaying, we print "85.25" because we do `ToString("0.00")` which will do normal rounding. 

Same applies to below calculation for branch & method.

Reply:
Vishnu is this comment out of date?

Reply:
Yes.. Since we are rounding it in property `AverageModulePercent ` itself, we are good now.